File: Java/BinarySearch.java
import java.util.Scanner;

// works only on sorted lists/arrays

public class BinarySearch {
    public static int binarySearch(int[] arr, int target) {
        int left = 0;
        int right = arr.length - 1;

        while(left <= right) {
            int mid = left + (right - left) / 2;

            if(arr[mid] == target) return target;

            if(arr[mid] < target) {
                left = mid + 1;
            } else {
                right = mid - 1;
            }
        }
        return -1;
    }

    public static void main(String[] args) {
        int arr[] = {1, 2, 3, 4, 5, 6, 7, 8, 9};

        Scanner sc = new Scanner(System.in);
        System.out.println("enter element to search from 1 to 9: ");
        int target = sc.nextInt();

        int result = binarySearch(arr, target);
        if(result != -1) {
            System.out.println("element found at index: " + result);
        } else {
            System.out.println("element not found.");
        }
    }
}

File: Python/BinarySearch.py
# Works only on sorted lists/arrays

def binary_search(arr, target):
    left, right = 0, len(arr) - 1

    while left <= right:
        mid = left + (right - left) // 2

        if arr[mid] == target:
            return target
        elif arr[mid] < target:
            left = mid + 1
        else:
            right = mid - 1

    return -1

def main():
    arr = [1, 2, 3, 4, 5, 6, 7, 8, 9]

    target = int(input("Enter element to search from 1 to 9: "))
    result = binary_search(arr, target)

    if result != -1:
        print(f"Element found at index: {result}")
    else:
        print("Element not found.")

if __name__ == "__main__":
    main()